### DISCUSSION TITLE
k-fold cross-validation in AutoGluon

### DISCUSSION THREAD

**Discussion by acbrandaoa1** [autogluon/autogluon#2185] [Q&A] [answered]
Does AutoGluon perform k-fold cross-validation when the goal is to deploy just a single model?

If yes, how can one set the number of folds?

Any help would be much appreciated...

**Comment by FailedPipeline** [comment]


**Reply by xinlnix** [reply]
I suppose @acbrandaoa1 means k-fold cross-validation, instead of bagging folds. Is the ` num_bag_folds` used for bagging not validate? If I wrong, please point out it. Thanks.

**Reply by FailedPipeline** [reply]
@xinlnix I think you're right and so I deleted my reply thank you.

**Comment by Innixma** [comment, marked as answer]
`num_bag_folds` and `num_bag_sets` control this. Use `refit_full=True` to collapse bags into single models for deployment purposes. Refer to TabularPredictor's [documentation](https://auto.gluon.ai/stable/api/autogluon.predictor.html#module-0) for more details.

**Comment by acbrandaoa1** [comment]
Hello.

Thank you for your response.

Regards.
